connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaConfigBackingStore.java

```diff
@@ -733,10 +734,11 @@ public void onCompletion(Throwable error, ConsumerRecord<String, byte[]> record)
                         new SecretKeySpec(key, (String) keyAlgorithm),
                         (long) creationTimestamp
                     );
-
-                    if (started)
-                        updateListener.onSessionKeyUpdate(KafkaConfigBackingStore.this.sessionKey);
```

Comment:
`KafkaConfigBackingStore.this.sessionKey` is already a `volatile` member variable. 

So, given that no other common data structures are altered in this block, I'm suspecting that your change means that we might as well get rid off the `synchronized` block altogether in this `else if` branch here. 

I'd like to give it a second thought in the morning but lmk what you think.

Reply:
Hi, @kkonstantine !
Yes, indeed, it looks like we can get rid off the synchronized block here at all. I think it's safe.
I will commit this change.